HIP/hip-xxx.md

```diff
@@ -0,0 +1,167 @@
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/
+updated:
+---
+
+## Abstract
+
+Hedera customers have requested the ability to view their current NFT allowances. Hence, we propose adding a
+new mirror node REST API `/api/v1/accounts/{id}/allowances/nfts` to return this information.
+
+## Motivation
+
+NFT marketplaces have requested the ability to view a NFT allowance's details via an API.
+
+## Rationale
+There have been requests to enable querying spending approvals for NFTs based on the owner ID. Additionally, there is a request to be able to query by the spender ID as well. It was also asked for the ability to know whether the approval was given via `approveForAll` or not.
+
+The API described below has been chosen to make the implementation easier for NFT explorers and marketplaces that wish to search NFT allowances based on the owner and spender.The decision to design one API for both use cases is aimed at reducing the number of APIs that essentially perform similar tasks. This API would also return the `approveForAll` information.
+
+## User stories
+
+- As a mirror node operator, I want to provide my API clients with the ability to list NFT allowances for an account that has granted allowance for a specific NFT serial number, as well as for those who have used  `approveForAll`.
+- As a mirror node operator, I want to provide my API clients with the ability to list NFT allowances for a spender who has received an allowance for a single NFT serial number, as well as via `approveForAll`.
+
+## Specification
+
+This is a HIP for adding the ability to query the allowances on NFTs from the mirror node.
+
+### Constraints
+
+When designing the NFT allowances APIs, it is important to consider certain constraints on the number of allowances and tokens. These constraints may limit the ability to query unbounded information.
+
+- An approval transaction can include a maximum of 20 allowance change operations.
+- An account can have a maximum of 100 hbar, token, and approved for all NFT allowances.
+- An account can have an unbounded number of individual NFT allowances.
+- A non-fungible token can have an unbounded number of approved for all NFT allowances.
+- A specific NFT can only have one explicit approval. The last one to be approved will take precedence.
+- When an NFT is transferred by either the owner or a spender, it results in the individual spender's allowance being cleared.
+- A specific NFT can have an unbounded number of implicit approved for all NFT allowances.
+
+### Approved for all NFT Allowances API
+
+This API takes a path parameter that represents either the owner or spender, depending on a boolean flag provided as a query parameter called `owner`. If the `owner` value is true, the `accountId` path parameter should indicate the ID of the owner. Conversely, if the `owner` value is false, the `accountId` path parameter should indicate the ID of the spender who has an allowance.
```

Comment:
Need to talk about how the response changes depending upon the owner flag. When owner is true it will return the allowances the owner has granted to various spenders but when false it will return the allowances granted to the spender by various owners of those tokens.